core/kazoo_ledgers-1.0.0/src/kz_ledger.erl

```diff
@@ -0,0 +1,83 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2011-2015, 2600HZ, INC
+%%% @doc
+%%%
+%%% @end
+%%% @contributors
+%%% Peter Defebvre
+%%%-------------------------------------------------------------------
+-module(kz_ledger).
+
+-include("kzl.hrl").
+
+-export([get/2]).
+-export([credit/3, credit/4, debit/3, debit/4]).
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%%
+%% @end
+%%--------------------------------------------------------------------
+-spec get(ne_binary(), ne_binary()) -> {'ok', integer()} | {'error', _}.
+get(Account, Name) ->
+    Options = [
+        'reduce'
+        ,{'key', Name}
+    ],
+    case kazoo_modb:get_results(Account, ?LIST_BY_TYPE, Options) of
+        {'ok', []} -> {'ok', 0};
+        {'error', _R}=Error -> Error;
+        {'ok', [JObj|_]} ->
+            {'ok', wh_json:get_integer_value(<<"value">>, JObj, 0)}
+    end.
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%%
+%% @end
+%%--------------------------------------------------------------------
+-spec credit(ne_binary(), ne_binary(), integer()) -> {'ok', ledger()} | {'error', _}.
+-spec credit(ne_binary(), ne_binary(), integer(), api_binary()) -> {'ok', ledger()} | {'error', _}.
+credit(Name, Amount, Account) ->
+    credit(Name, Amount, Account, 'undefined').
+
+credit(Name, Amount, Account, Desc) ->
+    create(<<"credit">>, Name, Amount, Account, Desc).
+
```

Comment:
I also strongly dislike adding function arguments to the beginning of the args list. If you have `f(A, B, C) -> ...` it should call `f(A, B, C, D)` not `f(D, A, B, C)`.